### PR TITLE
Re-enabling IL re-writing for System.Slices.

### PR DESCRIPTION
**Pull request by livarcocc**
Added the steps to do the re-writing as a post compile script.

### CODE REVIEW

**Comment by livarcocc** [comment]
After we have a good System.Slices package in myget, we can get the other tests passing as well and I can re-enable the tests in the CI. This was enough for System.Slices tests to pass though.

**Comment by adamsitnik** [comment]
I was just wondering how people that did PR in the time of "no rewrite scripts" managed to run unit tests ;) Very desired commit, at least for me.

**Comment by KrzysztofCwalina** [comment]
looks good.

**Comment by KrzysztofCwalina** [comment]
@adamsitnik, the tests were running against the previous version of System.Slices that was on myget. This is why I don't think we should be using nuget for cross-project references in the same repo.

File: src/System.Slices/src/project.json [hunk 1]
```diff
@@ -23,5 +23,8 @@
   },
   "frameworks": {
     "dotnet5.1": { }
+  },
+  "scripts": {
+    "postcompile": "re_write_il.cmd %compile:OutputDir% %compile:OutputFile%"
```
Comment (davidfowl) [inline src/System.Slices/src/project.json]: nice

**Comment by adamsitnik** [comment]
@KrzysztofCwalina Dzięki!

**Comment by KrzysztofCwalina** [comment]
Translating: "thanks!" :-)

**Comment by omariom** [comment]
@livarcocc 

It doesn't rewrite for me.
When try to start ilasm.exe manually  it says that it can't find fusion.dll.
Is it something with my machine?

Here is the output:
```
1>------ Rebuild All started: Project: System.Slices, Configuration: Debug Any CPU ------
1>  Compiling System.Slices for .NETPlatform,Version=v5.1
1>
1>  C:\DEV\GIT\Forks\corefxlab\src\System.Slices\src>..\tools\ildasm.exe /linenum /out:"C:\DEV\GIT\Forks\corefxlab\nuget\Release\dotnet5.1"\system.slices.beforerewrite.il /nobar "C:\DEV\GIT\Forks\corefxlab\nuget\Release\dotnet5.1\System.Slices.dll"
1>
1>  C:\DEV\GIT\Forks\corefxlab\src\System.Slices\src>..\tools\ILSub\ILSub.exe "C:\DEV\GIT\Forks\corefxlab\nuget\Release\dotnet5.1"\system.slices.beforerewrite.il "C:\DEV\GIT\Forks\corefxlab\nuget\Release\dotnet5.1"\system.slices.rewritten.il
1>
1>  C:\DEV\GIT\Forks\corefxlab\src\System.Slices\src>..\tools\ilasm.exe /quiet /pdb /dll /out:"C:\DEV\GIT\Forks\corefxlab\nuget\Release\dotnet5.1\System.Slices.dll" /nologo "C:\DEV\GIT\Forks\corefxlab\nuget\Release\dotnet5.1"\system.slices.rewritten.il
1>
1>  Compilation succeeded.
1>      0 Warning(s)
1>      0 Error(s)
1>
1>  Time elapsed 00:00:06.1656039
1>   (The compilation time can be improved. Run "dotnet build --build-profile" for more information)
1>
1>  Producing nuget package "System.Slices.0.1.0-e160206-1" for System.Slices
1>  System.Slices -> C:\DEV\GIT\Forks\corefxlab\nuget\Release\System.Slices.0.1.0-e160206-1.nupkg
1>  Producing nuget package "System.Slices.0.1.0-e160206-1.symbols" for System.Slices
1>  System.Slices -> C:\DEV\GIT\Forks\corefxlab\nuget\Release\System.Slices.0.1.0-e160206-1.symbols.nupkg
2>------ Rebuild All started: Project: System.Slices.tests, Configuration: Debug Any CPU ------
2>  Compiling System.Slices for .NETPlatform,Version=v5.1
2>
2>  C:\DEV\GIT\Forks\corefxlab\src\System.Slices\src>..\tools\ildasm.exe /linenum /out:"C:\DEV\GIT\Forks\corefxlab\src\System.Slices\src\bin\Release\dotnet5.1"\system.slices.beforerewrite.il /nobar "C:\DEV\GIT\Forks\corefxlab\src\System.Slices\src\bin\Release\dotnet5.1\System.Slices.dll"
2>
2>  C:\DEV\GIT\Forks\corefxlab\src\System.Slices\src>..\tools\ILSub\ILSub.exe "C:\DEV\GIT\Forks\corefxlab\src\System.Slices\src\bin\Release\dotnet5.1"\system.slices.beforerewrite.il "C:\DEV\GIT\Forks\corefxlab\src\System.Slices\src\bin\Release\dotnet5.1"\system.slices.rewritten.il
2>
2>  C:\DEV\GIT\Forks\corefxlab\src\System.Slices\src>..\tools\ilasm.exe /quiet /pdb /dll /out:"C:\DEV\GIT\Forks\corefxlab\src\System.Slices\src\bin\Release\dotnet5.1\System.Slices.dll" /nologo "C:\DEV\GIT\Forks\corefxlab\src\System.Slices\src\bin\Release\dotnet5.1"\system.slices.rewritten.il
2>
2>  Compilation succeeded.
2>      0 Warning(s)
2>      0 Error(s)
2>
2>  Time elapsed 00:00:06.3815430
2>  Compiling System.Slices.Tests for DNXCore,Version=v5.0
2>
2>  Compilation succeeded.
2>      0 Warning(s)
2>      0 Error(s)
2>
2>  Time elapsed 00:00:06.2067403
2>   (The compilation time can be improved. Run "dotnet build --build-profile" for more information)
2>
2>  Producing nuget package "System.Slices.Tests.1.0.0" for System.Slices.Tests
2>  System.Slices.Tests -> C:\DEV\GIT\Forks\corefxlab\nuget\Release\System.Slices.Tests.1.0.0.nupkg
2>  Producing nuget package "System.Slices.Tests.1.0.0.symbols" for System.Slices.Tests
2>  System.Slices.Tests -> C:\DEV\GIT\Forks\corefxlab\nuget\Release\System.Slices.Tests.1.0.0.symbols.nupkg
========== Rebuild All: 2 succeeded, 0 failed, 0 skipped ==========

```

**Comment by buybackoff** [comment]
I have found an issue that it works only with `dotnet build`, doesn't work inside Visual Studio build. 

@omariom Ihave the same message about fusion.dll but it doesn't affect the rewriting via .cmd file.

**Comment by omariom** [comment]
@buybackoff 
It was fixed by https://github.com/dotnet/corefxlab/pull/662
I can't remember the last time  I built it in VS. It is easier to build just Slices with `dotnet build`

**Pull request merged**